### DISCUSSION TITLE
2D Autocorrelation behavior

### DISCUSSION THREAD

**Discussion by edwardclem** [rdkit/rdkit#3790] [Q&A]
Hi everyone!

I've been testing some of the descriptors in `rdkit.Chem.Descriptors` and found a behavior that may or may not be a bug - for some short carbon chains (`CCCC` for example) some of the 2D autocorrelation descriptors report nans. 

My understanding of these descriptors is that they represent autocorrelations between pairs of atoms at specified distances, using some atomic property as a weighting function (6 properties are specified in the code, so I'm assuming the 192-dimensional vector refers to these 6 weights at distances up to 24 bonds?). It makes sense that some of these descriptors wouldn't be applicable to this molecule, as there aren't any paths longer than 4, but I'm not sure why they would report nans unless there's some normalization issue happening somewhere. Is this an expected behavior?

Steps to reproduce:

```
from rdkit import Chem
from rdkit.Chem.rdMolDescriptors import CalcAUTOCORR2D
mol = Chem.MolFromSmiles("CCCC")
print(CalcAUTOCORR2D(mol))
```

I've experienced this using rdkit 2020.09.03 on Ubuntu 18.04 and macOS 10.14. The same thing also appears when calling the individual autocorrelation descriptor functions instead of the `CalcAUTOCORR2D` function.

**Comment by greglandrum** [comment]
Hi @edwardclem,
this seems to be a bug in the AUTOCORR2D descriptor calculator.
We'll take a look.
